internal/node/node.bzl

```diff
@@ -34,11 +40,27 @@ def _write_loader_script(ctx):
         escaped = mn.replace("/", r"\/").replace(".", r"\.")
         mapping = r"{module_name: /^%s\b/, module_root: '%s'}" % (escaped, mr)
         module_mappings.append(mapping)
-  workspace = ctx.attr.node_modules.label.workspace_root.split("/")[1] if ctx.attr.node_modules.label.workspace_root else ctx.workspace_name
-  node_modules_root = "/".join([f for f in [
-      workspace,
-      ctx.attr.node_modules.label.package,
-      "node_modules"] if f])
+  node_modules_root = None
+  if ctx.attr.node_modules:
+    workspace = ctx.attr.node_modules.label.workspace_root.split("/")[1] if ctx.attr.node_modules.label.workspace_root else ctx.workspace_name
+    node_modules_root = "/".join([f for f in [
+        workspace,
+        ctx.attr.node_modules.label.package,
+        "node_modules"] if f])
+  for d in ctx.attr.data:
+    if NodeModuleInfo in d:
+      possible_root = "/".join([d[NodeModuleInfo].workspace, "node_modules"])
+      if not node_modules_root:
+        node_modules_root = possible_root
+      elif node_modules_root != possible_root:
+        fail("All npm dependencies need to come from a single workspace. found", [node_modules_root, possible_root])
+  if not node_modules_root:
+    fail("""
+         Due to a breaking change in rules_nodejs, target %s
+         must now declare either an explicit node_modules attribute, or
+         list explicit deps[] or data[] dependencies on npm labels.
+
+         See http://FIXME""" % ctx.label)
```

Comment:
@alexeagle Would be good not to introduce this breaking change. Could we detect if the node_modules was not set and there were no fine grained deps and use the default of `//:node_modules` in that case?

Reply:
Hmmm. I see what the reasoning is. We don't want to force user's to defined `//:node_modules` filegroup is the move to fine grained deps. rollup_bundle will also need to be updated to removed the `@//:node_modules` default:
```
    "node_modules": attr.label(
        doc = """Dependencies from npm that provide some modules that must be resolved by rollup.""",
        default = Label("@//:node_modules")),
```
Where were you thinking the `http://FIXME` would point?

Reply:
the fixme should be a page on the wiki that explains the change and walk through how to do the upgrade